### ISSUE TITLE
Results processing could potentially be sped up

### ISSUE THREAD

**Issue by jokochems** [oemof/oemof-solph#794]
## Status quo / issue 
The current workflow in the processing(.py) module is roughly as follows:

- Results obtained from pyomo are first stored in an overall pandas.DataFrame.
- Afterwards, this DataFrame is split up by components and stored in a dictionary of DataFrames for each component.
- Then, there is an iteration over the dict of DataFrames and the results are stored in a common nested dictionary format with the components as keys for the outer dict and "scalars" and "sequences" as keys for the inner dict.

While I think, this is quite intuitive, it might bring along some performance issues for large models. We have noticed quite some computational effort for converting results of large simulation runs. 

## Idea to tackle / discuss
I think the conversion step in between is not needed and maybe, one could use vectorization in pandas or numpy instead of the dict data structures.

I have not yet thought it through completely, esp. not for other consequences this might bring along, but I think, this could be a thing to discuss in the upcoming dev meeting https://github.com/oemof/oemof/issues/96

**Comment by p-snft** [comment]
Could a new implementation also solve the dumping problem (#781)?

**Comment by uvchik** [comment]
No, @p-snft that seems to be different because in #781 the user wanted to dump the EnergySystem before(!) the optimisation.

**Comment by Bachibouzouk** [comment]
Hello,

In a project of RLI we are using a Dataframe structure which we first built form the DataFrame dict but after @p-snft updated this issue, I realised we could also directly create it from the pyomo DataFrame. The structure of our DataFrame is defined in this discussion thread: https://github.com/rl-institut/oemof-tabular-plugins/discussions/10 

The reason why we created oemof-tabular-plugins was because we needed to be able to do releases quickly to meet project deadlines. However, our wish is to move most of the code we wrote there into oemof-tabular (for the preprocessing part) and  to oemof-solph (for the postprocessing part). Having an output of oemof results in a Datapackage format (similar to the structure of of oemof-tabular datapackage) would be ideal for us. I am not sure what is the best way forward. I just know that there are some functionnalities of oemof-solph I do not currently use and therefore I am not at the best place to ensure my solution for oemof-result cover those cases. I would be interested to participate into a discussion regarding an alternative structure of oemof-solph output :)

**Comment by fwitte** [comment]
<del>Is this topic still open? Maybe it could also be interesting to look into lazy loading of the results...</del>

Thanks to @p-snft for quickly letting me know: already inbuilt :)

**Comment by p-snft** [comment]
Indeed, #1181 implemented this idea.
